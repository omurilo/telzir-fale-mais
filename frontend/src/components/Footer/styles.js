import styled, { css } from "styled-components";
import { Facebook, Twitter } from "styled-icons/boxicons-logos";

export const Footer = styled.footer`
  ${({ theme }) => css`
    padding: ${theme.spacing.default};
    display: flex;
    justify-content: space-between;
    align-items: center;
  `}

  & > div {
    display: flex;
    align-items: center;

    & > svg {
      color: ${({ theme }) => theme.colors.white};
      background: ${({ theme }) => theme.colors.primary};
      border-radius: 50%;
      padding: ${({ theme }) => theme.spacing.little};
      margin-left: ${({ theme }) => theme.spacing.tiny};
    }
  }
`;

export const FaceIcon = styled(Facebook)``;

export const TwitterIcon = styled(Twitter)``;
