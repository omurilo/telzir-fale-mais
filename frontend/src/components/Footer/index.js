import React from "react";

import Logo from "components/Logo";

import * as Styled from "./styles";

const Footer = () => {
  return (
    <Styled.Footer>
      <Logo />
      <div>
        <Styled.FaceIcon size="32" />
        <Styled.TwitterIcon size="32" />
      </div>
    </Styled.Footer>
  );
};

export default Footer;
