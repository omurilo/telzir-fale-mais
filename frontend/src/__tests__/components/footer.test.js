import React from "react";
import Footer from "components/Footer";

import { render, screen } from "../test-utils";

describe("Footer", () => {
  it("should render the Footer", () => {
    render(<Footer />);

    expect(screen.getByRole("contentinfo")).toMatchSnapshot();
  });
});
