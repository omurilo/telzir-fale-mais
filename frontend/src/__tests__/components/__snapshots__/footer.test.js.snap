// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render the Footer 1`] = `
<footer
  class="styles__Footer-sc-7cxnjx-0 cHgCGm"
>
  <div>
    <span
      class="styles__LogoItem-isvygu-0 jiqBiG"
    >
      T
    </span>
    <span
      class="styles__LogoItem-isvygu-0 jiqBiG"
    >
      E
    </span>
    <span
      class="styles__LogoItem-isvygu-0 jiqBiG"
    >
      L
    </span>
    <span
      class="styles__LogoItem-isvygu-0 iUeJxp"
      color="secondary"
      rotate="25"
      scale="1.25"
    >
      Z
    </span>
    <span
      class="styles__LogoItem-isvygu-0 jiqBiG"
    >
      I
    </span>
    <span
      class="styles__LogoItem-isvygu-0 jiqBiG"
    >
      R
    </span>
  </div>
  <div>
    <svg
      aria-hidden="true"
      class="StyledIconBase-ea9ulj-0 llIrRb styles__FaceIcon-sc-7cxnjx-1"
      fill="currentColor"
      focusable="false"
      height="32"
      viewBox="0 0 24 24"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13.397 20.997v-8.196h2.765l.411-3.209h-3.176V7.548c0-.926.258-1.56 1.587-1.56h1.684V3.127A22.336 22.336 0 0014.201 3c-2.444 0-4.122 1.492-4.122 4.231v2.355H7.332v3.209h2.753v8.202h3.312z"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="StyledIconBase-ea9ulj-0 llIrRb styles__TwitterIcon-sc-7cxnjx-2"
      fill="currentColor"
      focusable="false"
      height="32"
      viewBox="0 0 24 24"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19.633 7.997c.013.175.013.349.013.523 0 5.325-4.053 11.461-11.46 11.461-2.282 0-4.402-.661-6.186-1.809.324.037.636.05.973.05a8.07 8.07 0 005.001-1.721 4.036 4.036 0 01-3.767-2.793c.249.037.499.062.761.062.361 0 .724-.05 1.061-.137a4.027 4.027 0 01-3.23-3.953v-.05c.537.299 1.16.486 1.82.511a4.022 4.022 0 01-1.796-3.354c0-.748.199-1.434.548-2.032a11.457 11.457 0 008.306 4.215c-.062-.3-.1-.611-.1-.923a4.026 4.026 0 014.028-4.028c1.16 0 2.207.486 2.943 1.272a7.957 7.957 0 002.556-.973 4.02 4.02 0 01-1.771 2.22 8.073 8.073 0 002.319-.624 8.645 8.645 0 01-2.019 2.083z"
      />
    </svg>
  </div>
</footer>
`;
